TestCPP11.cmake

```diff
@@ -98,6 +100,13 @@ while(TESTS_CPP11)
 
   if(CXX_${TEST_CPP11_NAME}_SUPPORTED)
     add_definitions(-DCXX_${TEST_CPP11_NAME}_SUPPORTED)
+    list(APPEND __supported ${TEST_CPP11_name})
+  else()
+    list(APPEND __unsupported ${TEST_CPP11_name})
   endif()
 endwhile()
 
+if(__unsupported)
+  set(__unsupported " UNSUPPORTED: ${__unsupported}")
+endif()
+message(STATUS "Tested C++11 features: ${__supported}${__unsupported}")
```

Comment:
Probably that's something more in line what you're looking for.

In general, the usefulness of the TestCPP11 is quite limited nowadays. All the features we check are present in all the compilers we use, and even if a certain feature is not present, I don't remember seeing any handling in that case.

CMake 3 solved that problem (finally), by specifying the required features on the target: http://www.cmake.org/cmake/help/v3.2/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html#prop_gbl:CMAKE_CXX_KNOWN_FEATURES

If we dare, we retire this TestCPP11 script, use the cmake feature for CMake 3 and CMake 2 does not perform any check.